server/src/main/java/io/crate/expression/operator/any/AnyOperators.java

```diff
@@ -22,42 +22,73 @@
 
 package io.crate.expression.operator.any;
 
+import io.crate.common.collections.Lists2;
+import io.crate.expression.operator.LikeOperators;
 import io.crate.expression.operator.OperatorModule;
 import io.crate.sql.tree.ComparisonExpression;
 
 import java.util.Arrays;
 import java.util.Collection;
+import java.util.List;
 import java.util.Locale;
+import java.util.function.IntPredicate;
 
 import static io.crate.expression.operator.any.AnyOperator.OPERATOR_PREFIX;
-import static io.crate.expression.operator.any.AnyOperators.Names.EQ;
-import static io.crate.expression.operator.any.AnyOperators.Names.GT;
-import static io.crate.expression.operator.any.AnyOperators.Names.GTE;
-import static io.crate.expression.operator.any.AnyOperators.Names.LT;
-import static io.crate.expression.operator.any.AnyOperators.Names.LTE;
-import static io.crate.expression.operator.any.AnyOperators.Names.NEQ;
 
 public final class AnyOperators {
 
-    private AnyOperators() {
-    }
+    public static final List<String> OPERATOR_NAMES = Lists2.concat(
+        AnyOperators.Type.fullQualifiedNames(),
+        List.of(
+            LikeOperators.ANY_LIKE,
+            LikeOperators.ANY_ILIKE,
+            LikeOperators.ANY_NOT_LIKE,
+            LikeOperators.ANY_NOT_ILIKE
+        )
+    );
 
     public static class Names {
-        public static final String EQ = OPERATOR_PREFIX + ComparisonExpression.Type.EQUAL.getValue();
-        public static final String GTE = OPERATOR_PREFIX + ComparisonExpression.Type.GREATER_THAN_OR_EQUAL.getValue();
-        public static final String GT = OPERATOR_PREFIX + ComparisonExpression.Type.GREATER_THAN.getValue();
-        public static final String LTE = OPERATOR_PREFIX + ComparisonExpression.Type.LESS_THAN_OR_EQUAL.getValue();
-        public static final String LT = OPERATOR_PREFIX + ComparisonExpression.Type.LESS_THAN.getValue();
-        public static final String NEQ = OPERATOR_PREFIX + ComparisonExpression.Type.NOT_EQUAL.getValue();
+        public static final String EQ = Type.EQ.fullQualifiedName;
+        public static final String GTE = Type.GTE.fullQualifiedName;
+        public static final String GT = Type.GT.fullQualifiedName;
+        public static final String LTE = Type.LTE.fullQualifiedName;
+        public static final String LT = Type.LT.fullQualifiedName;
+        public static final String NEQ = Type.NEQ.fullQualifiedName;
+    }
+
+    public enum Type {
```

Comment:
Is this to assign the name with the predicate? (Just to make sure I understand why this is being changed)

I'm not sure I see the benefit of doing this in this new way?

Reply:
We need to match inside the rule predicates again a list of all full-qualified internal operator names. (https://github.com/crate/crate/pull/10076/files#diff-cc8e036acde8edb2dfa2dd74f78d3336R40) 
Also by doing it this way we just can loop over the enum to register the operators.
We could also do this in a concrete way (like it was before) and additionally store a list of the operator names. I guess this is a bit of personal preference, no strong opinion here.

Reply:
> We need to match inside the rule predicates again a list of all full-qualified internal operator names

Ah I missed that.

Could we now also remove the `Names` class and refer to the enum directly instead?